tests/unit/data/test_storage.py

```diff
@@ -278,3 +285,204 @@ def test_all_data_cached(self):
 
         # noinspection PyUnresolvedReferences
         ds.data_file.get.assert_called_once()
+
+
+class DataSetStoreTest(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.mock_fs = Mock(spec=S3FileSystem)
+        self.mock_fs.anon = False
+        self.mock_storage_bucket_name = "test_bucket"
+
+        test_metadata = Metadata("test dataset", box(-2, -2, 2, 2))
+
+        self.mock_metadata_store = Mock(spec=S3FSMetadataStore)
+        self.mock_metadata_store.get.return_value = test_metadata
+
+        self.dataset_store = S3FSDataSetStore(self.mock_fs, self.mock_metadata_store, self.mock_storage_bucket_name)
+
+        # This is a little bit circular, but makes it easier to set up tests for download operations
+        # later, since the paths will match
+        data_dir = Path(self.dataset_store._cache_dir) / Path(test_metadata.id)
+        data_files = [data_dir / p for p in [Path("test-file.csv")]]
+        self.test_dataset = DataSet(data_files, test_metadata)
+
+    def test_available_datasets(self):
+        """
+        WHEN available_datasets is called
+        THEN a list of datasets that exist in the storage bucket is returned
+        """
+        expected_datasets = ["dataset1", "dataset2", "dataset3"]
+        self.mock_fs.ls.return_value = [f"{self.mock_storage_bucket_name}/{ds}" for ds in expected_datasets]
+
+        actual_datasets = self.dataset_store.available_datasets()
+        self.assertEqual(expected_datasets, actual_datasets)
+
+    def test_get(self):
+        """
+        GIVEN a valid ID for a dataset that exists on the object store
+        WHEN get is called
+        THEN that dataset is returned
```

Comment:
It also ensures that only one test (assert) is included per test, which is something Barnaby and I have debated quite a lot.  We decided in the end that several asserts per test is preferable, but I never thought it was really. I just wanted to stop arguing about it.